### ISSUE TITLE
FFXVI Game Files replaced and Ant-tamper flag triggered

### ISSUE THREAD

**Issue by pyrotiger** [sonic2kk/steamtinkerlaunch#1205]
I applied SteamTinkerLaunch to FFXVI so i could use Proton-tkg as the comp tool, however it replaced my game's saved files and created symlinks. That triggered the antitamper flag with Deluga, forcing me to wait 24 hrs and i seem to have lost my game's saved files. Where was it moved so i can restore the game's save files?? Valve refuses to help in this matter(restore from a cloud save) because i'm using this app.

v12.12
Garuda running zen3 6.12 kernel
Preinstalled as part of Garuda Gamer Edition

[steamtinkerlaunch.log](https://github.com/user-attachments/files/18264753/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
SteamTinkerLaunch won't create symlinks afaik unless you have configured it to do so. Based on the log it does not appear that SteamTinkerLaunch did anything.

It sounds like something was changed with the prefix, as I assume that's where the game files are stored.

I don't own FFXVI so I can't say for sure what happened here. However what may have happened is that you launched the game with Proton-tkg, which re-created the prefix. When prefixes get upgraded from one version, say Proton 9.0, to Proton-tkg, based on a much newer Wine version, sometimes they have to be re-created. This could have been what happened, and it wouldn't be specific to SteamTinkerLaunch. This often also results in save data loss (As an example, I have lost Sonic Adventure 2 saves as a result of this).

Without more information or someone being able to reproduce this with SteamTinkerLaunch consistently, I can't say for sure what caused this. If you launched the game with a Proton version before using Proton-tkg, it is possible that the prefix had to be re-created.

<hr>

Though it came pre-installed, it should be noted that v12.12 is very out-of-date, almost 2 years out-of-date compared to latest master.

**Comment by pyrotiger** [comment]
v12.12 game of steiner is whats in the master? to be precise i have 12.12-1.2 installed. Yes i see that what is posted in the master is very old, dated 2023. 

That sounds accurate in what happened, i recall that the game had to redo its wine prefix when i launched FFXVI after switching to SteamTinkerLaunch. However i am able to switch between provided proton versions within Steam without running across this issue. I tested STL with Star Trek Online, no issue with saved game file as its stored on the server side nowadays, but it did ask me to verify my account via 2FA as if I had just installed the game. Apparently a software token is stored locally for online games and when switching to SteamTinkerLaunch, that token is invalidated.

Fortunately i had tested out the game on my Windows side before trying STL and it had synced the save file there so I was able to recover my saved game minus the last 30 min of gameplay. Steam customer support refuses to restore/rollback from an online backup copy, even tho the feature is there. ::rme::

**Comment by sonic2kk** [comment]
> v12.12 game of steiner is whats in the master? to be precise i have 12.12-1.2 installed. Yes i see that what is posted in the master is very old, dated 2023.

No, that is not master. That is the last release, from March 2023. What's in master, at the time our comments were posted, was from October 25th 2024. If you compare v12.12 with what's in master, you can see there are [quite a lot of changes](https://github.com/sonic2kk/steamtinkerlaunch/compare/v12.12...master) (273 commits).

> However i am able to switch between provided proton versions within Steam without running across this issue.

It could happen for any reason with any tool. I'm not sure how Proton determines it, I haven't combed through the Python script, but it could be that whatever reason the Proton script found to modify the prefix just happened to coincide with using SteamTinkerLaunch.

> Apparently a software token is stored locally for online games and when switching to SteamTinkerLaunch, that token is invalidated.

Does this only happen with SteamTinkerLaunch or does this also happen if you were to use, say, Proton 9.0-4, then switch to Proton-tkg? I don't see why this would be specific to using SteamTinkerLaunch, since it should run the same launch command that Steam does when it ultimately launches the game with the selected compatibility tool.

> Steam customer support refuses to restore/rollback from an online backup copy, even tho the feature is there

For the record, SteamTinkerLaunch is unaffiliated with Valve and Steam. As noted on the [Readme](https://github.com/sonic2kk/steamtinkerlaunch?tab=readme-ov-file#disclaimer) (emphasis by me):

> Keep in mind that **you are using SteamTinkerLaunch at your own risk** and that you are responsible for the third party programs that you launch with it. SteamTinkerLaunch is **not affiliated with Valve Corporation or Steam**.

<hr>

This does not sound like an issue specific to SteamTinkerLaunch, but more a consequence of switching Proton versions and the impact this can have on a prefix.

**Comment by sonic2kk** [comment]
Closing as there have been no further details provided for a while and this was an issue encountered with v12.12. It sounds like a general DRM issue.

If there are issues found generally with SteamTinkerLaunch and DRM they can be investigated in future.
